google-cloud-firestore/src/main/java/com/google/cloud/firestore/VectorQuery.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.Timestamp;
+import com.google.firestore.v1.RunQueryRequest;
+import com.google.firestore.v1.StructuredQuery;
+import com.google.protobuf.ByteString;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * A query that finds the documents whose vector fields are closest to a certain query vector.
+ * Create an instance of `VectorQuery` with {@link Query#findNearest}.
+ */
+public final class VectorQuery extends StreamableQuery<VectorQuerySnapshot> {
+  final Query query;
+  final FieldPath vectorField;
+  final VectorValue queryVector;
+  final int limit;
+  final DistanceMeasure distanceMeasure;
+  final VectorQueryOptions options;
+
+  /** Creates a VectorQuery */
+  VectorQuery(
+      Query query,
+      FieldPath vectorField,
+      VectorValue queryVector,
+      int limit,
+      DistanceMeasure distanceMeasure,
+      VectorQueryOptions options) {
+    super(query.rpcContext, query.options);
+
+    this.query = query;
+    this.options = options;
+    this.vectorField = vectorField;
+    this.queryVector = queryVector;
+    this.limit = limit;
+    this.distanceMeasure = distanceMeasure;
+  }
+
+  /**
+   * Executes the query and returns the results as {@link QuerySnapshot}.
+   *
+   * @return An ApiFuture that will be resolved with the results of the VectorQuery.
+   */
+  @Override
+  public ApiFuture<VectorQuerySnapshot> get() {
```

Comment:
I think we still want this to be `@Nonnull` ?

Reply:
same